### ISSUE TITLE
[Feature]: allow to set validationBehavior on a form (apply to all fields)

### ISSUE THREAD

**Issue by gabimor** [airjp73/remix-validated-form#184]
### What is the new or updated feature that you are suggesting?

I have a scenario where I want my form to start show error messages only after submit. 
this form has several fields, right now it seems that the only way to support this is to add `validationBehavior` prop to every field in the form. it would be nice to have an option to set a form-wide default. 

### Why should this feature be included?

DRY :)

**Comment by airjp73** [comment]
Hi!

Wouldn't that be _more_ duplication rather than less? You only have to define your `validationBehavior` once for each type of input you have (assuming you have [components for integrating with the lib](https://www.remix-validated-form.io/integrate-your-components)). If you put it at the form level, then you have to define it for every form in your app.

If we added this to the form level, it wouldn't be super obvious what should happen if you configure it at the form _and_ the field level. My usual instinct is that field-level config would override the form-level config, but really the field-level config is usually going to be global (e.g. your Select component has one setting and your Input has another).

I think putting this configuration at the `useField` level puts it in a good spot. If you're worried about being DRY (often overrated 😉), you could create a custom `useField` hook that applies those settings.

**Comment by gabimor** [comment]
Hi! Thanks for taking the time address this. Actually since writing I realized that my real scenario was this:
1. I have a modal that contains a validated field. 
2. when the modal opens the field received focus
3. when it closes it loses focus and thus the validation occurs, resulting in an ugly red text error that appears for a fraction of a second before the modal disappears.

so to workaround this I wanted to prevent this first field (the one that is focused when the modal opens) from validating on blur. but I now realize this isn't the right solution for this case. just wondering what would be a better one though

**Comment by airjp73** [comment]
Ah, so what you're looking for is more of a form-specific override? If changing the validationBehavior fixes the issue, can you add a prop to your input component that makes that change?

**Comment by gabimor** [comment]
Yeah that would work but what i realized is that I don't really want to change the field behavior, onBlur validation is great, just not whan the modal is closing :) but I'm wondering about a corrent solution to this issue since even clicking "cancel" on the form would cause a blur and trigger the validation. this seems more of a logical dilemma than a code related issue 😄
